### ISSUE TITLE
TreeGrid with height set to 100% causes missing entries on reload and flickering

### ISSUE THREAD

**Issue by focbenz** [vaadin/framework#10501]
If a TreeGrid with fixed width and height set to 100% some entries after a specific item start to flicker when the TreeGrid is reloaded and some entries might disappear. This issue has been following us since Vaadin 8 was introduced and can still be repoduced in the latest 8.1.7 or 8.2.0 Vaadin release.

![vaadin8missingnavigationentriesintreegrid](https://user-images.githubusercontent.com/23717010/34677181-90e86856-f48f-11e7-9e66-454a9effbc81.jpg)

We use the grid to show a navigation menu on the left side of the application as demonstrated the sample below

Sample project that reproduces the issue with a minimal setup:
https://github.com/FOCONIS/disappearing-nav-entries

Sample code:
```
                private TreeGrid<NavigationEntry> navigationTree = new TreeGrid<>();
(...)
		TreeDataProvider<NavigationEntry> dataProvider = new TreeDataProvider<>(data);
		navigationTree.setDataProvider(dataProvider);

		navigationTree.setWidth("250px");
		// Setting the height to 100% causes problem with flickering grid and disappearing items
		navigationTree.setHeight("100%");
		// Workaround: Use a fixed height for the grid and the flickering goes away
		//navigationTree.setHeight("700px");
```

**Comment by focbenz** [comment]
Please attack this issue ... very annoying issue since we use this element for our navigation.
Also we already tracked down the root cause ... but currently still need to use the 100% height.

**Comment by focbenz** [comment]
Vaadin Version 8.4.3 made the problem seriously worse ... whole parts of the navigation were missing.
Updating to latest 8.4.5 at least reduced the problem to the originally reported just "disappearing" single entries.

**Comment by focbenz** [comment]
We are not currently monitoring this issue anymore ... it might still be valid for other Vaadin users.

**Comment by focbenz** [comment]
There is sample code that could be turned into a Testbench test case ... if that is not enough the only option is to fix this issue yourself if you are also affected.

**Comment by focbenz** [comment]
Seems to be fixed in Vaadin 8.9.1

**Comment by TatuLund** [comment]
Thanks for your effort in testing. We did major fix in Escalator and expected it to fix some other issues as byproduct.
